components/hab/src/main.rs

```diff
@@ -1966,6 +1966,11 @@ fn ui() -> UI {
 /// Set all fields for an `SvcLoad` message that we can from the given opts. This function
 /// populates all *shared* options between `run` and `load`.
 fn svc_load_from_input(m: &ArgMatches) -> Result<sup_proto::ctl::SvcLoad> {
+    // TODO (DM): This check can eventually be removed.
+    // See https://github.com/habitat-sh/habitat/issues/7339
+    if m.is_present("APPLICATION") || m.is_present("ENVIRONMENT") {
+        ui().warn("--application and --environment flags are deprecated.")?;
```

Comment:
Also mentioning that they will be completely ignored would be a nice addition.

Reply:
Addressed in [20d99a](https://github.com/habitat-sh/habitat/pull/7335/commits/20d99aa36b4dfb14b5f615107863a756b9e18d41)